addons/base_vat/models/res_partner.py

```diff
@@ -61,13 +70,36 @@
     'tr': 'TR1234567890 (VERGINO) veya TR12345678901 (TCKIMLIKNO)'  # Levent Karakas @ Eska Yazilim A.S.
 }
 
+VIES_WSDL_URL = 'http://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl'
+
+VAT_CHECK_ANSWERS = [('verified', 'Verified'), ('wrong', 'Wrong'), ('unknown', 'Unknown')]
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
+    base_vat_check_status = fields.Selection(selection=VAT_CHECK_ANSWERS, string='VAT Verification Status', compute='compute_base_vat_check_status')
+    base_vat_vies_answer = fields.Selection(selection=VAT_CHECK_ANSWERS, string='Cached Answer from VIES', help="Technical field containing the answer received from VIES regarding this partner's VAT number, if the service has been contacted for it. None otherwise.")
+
+    @api.model
+    def compact_vat_number(self, vat):
+        """ Returns the vat number given in parameter, with all non-significative
+        characters removed, raising an excpetion if vat's format is incorrect.
+        If stdnum is not install, this function instead returns vat without
+        any blank or non-alphanumerical character.
+        """
+        if stdnum_vat: # We use stdnum if it is available, as it has individual compact functions applying the specificities of each country
+            return stdnum_vat.compact(vat)
+
+        return re.sub('\W', '', vat)
```

Comment:
```python
if stdnum_vat:
    try:
         # use stdnum if it is available, as it has individual compact functions applying the specificities 
         # of each country. Fallback on the naive compacting otherwise, or if it fails.
        return stdnum_vat.compact(vat)
    except ...........:
        pass
return re.sub('\W', '', vat)
```